enhancements/evidence-forwarding.md

```diff
@@ -0,0 +1,305 @@
+persistent storage:
+
+```yaml
+---
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSetting
+metadata:
+  name: default
+  namespace: openshift-compliance
+rawResultStorage:
+  pvAccessModes:
+  - ReadWriteOnce
+  rotation: 3
+  size: 1Gi
+roles:
+- worker
+- master
+scanTolerations:
+  default:
+  - operator: Exists
+  schedule: 0 1 * * *
+```
+
+A site reliability engineer could create another `ScanSetting` that bypasses
+storing evidence on physical volumes with the following:
+
+```yaml
+---
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSetting
+metadata:
+  name: external-storage
+  namespace: openshift-compliance
+externalResultStorage:
+  endpoint: https://object-storage.example.com/
+  tls:
+roles:
+- worker
+- master
+scanTolerations:
+  default:
+  - operator: Exists
+  schedule: 0 1 * * *
+```
+
+A site reliability engineer could configure both storage mechanisms for redundancy:
+
+```yaml
+---
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ScanSetting
+metadata:
+  name: external-storage-and-cluster-storage
+  namespace: openshift-compliance
+rawResultStorage:
+  pvAccessModes:
+  - ReadWriteOnce
+  rotation: 3
+  size: 1Gi
+externalResultStorage:
+  endpoint: https://object-storage.example.com/
+  tls:
+roles:
+- worker
+- master
+scanTolerations:
+  default:
+  - operator: Exists
+  schedule: 0 1 * * *
+```
+
+### User Stories
+
+1. As a site reliability engineer, I want to configure the Compliance Operator
+   to send all ARF reports to a dedicated object store.
+2. As a site reliability engineer, I want to opt out of storing any ARF reports
+   on physical volumes.
+3. As a site reliability engineer, I expect verbose alerts when the Compliance
+   Operator cannot send ARF reports to a configured external storage pool.
+4. As a site reliability engineer, I expect the Compliance Operator to fail if
+   the configured endpoint is unreachable.
+5. As a site reliability engineer, I expect to forward evidence to
+   public object-storage endpoints, like AWS S3.
+5. As a site reliability engineer, I expect to forward evidence to
+   private object-storage endpoints in cluster, Ceph or OpenShift Data Foundation.
+
+### API Extensions
+
+This proposal would extend the `ScanSetting` custom resource definition (CRD)
+to include an additional attribute called `externalResultStorage`.
+
+## `externalResultStorage`
+
+This attribute is an *optional* argument to `ScanSetting` CRDs. Existing
+instances of `ScanSetting` custom resources will be treated as if they're not
+configured to use external storage, making the feature backwards compatible.
+
+###  `endpoint`
+
+The `endpoint` is a required argument for `externalResultStorage`. The
+Compliance Operator will validate the `endpoint` at runtime (e.g., when a
+`ScanSetting` is created or used) to ensure connectivity. This doesn't
+necessarily mean the endpoint will always be available, but it is important to
+fail early if the URL is malformed.
+
+###  `tls`
```

Comment:
@mrogers950 left a bunch of good feedback on the results forwarding proposal that could probably be ported here, too.

Reply:
Im betting we need mTLS

Reply:
What if we're just dumping things in S3?

Reply:
That really depends on the receiver. Most S3 implementations do token-based authentication as far as I know.